lib/applications/resolvers.ts

```diff
@@ -6,14 +6,109 @@ import {
  * @returns All RCD applications
  */
-export const applications: Resolver = async (_parent, _args, { prisma }) => {
-  const applications = await prisma.application.findMany();
-  return applications;
+export const applications: Resolver = async (_parent, { filter }, { prisma }) => {
+  const { order, permitType, requestType, status, search, limit = 20, offset = 0 } = filter;
+
+  // Parse search string
+  let userIDSearch, firstSearch, middleSearch, lastSearch;
+
+  if (parseInt(search)) {
+    userIDSearch = parseInt(search);
+  } else if (search) {
+    [firstSearch, middleSearch, lastSearch] = search.split(' ');
+    middleSearch = middleSearch || firstSearch;
+    lastSearch = lastSearch || middleSearch;
+  }
+
+  // Parse sorting order
+  let orderBy = undefined;
+
+  if (order && order.length > 0) {
+    const sortingOrder: Array<Record<string, SortOrder>> = [];
+    order.forEach(([field, order]: [string, SortOrder]) => {
+      if (field === 'name') {
+        // Primary sort is by first name and secondary sort is by last name
+        sortingOrder.push({ firstName: order });
+        sortingOrder.push({ lastName: order });
+      } else if (field === 'dateReceived') {
+        sortingOrder.push({ createdAt: order });
+      }
+    });
+    orderBy = sortingOrder;
+  }
+
+  // Get number of applications with desired filters
+  const applicationsCount = await prisma.application.count({
+    where: {
+      applicant: {
+        id: userIDSearch,
+      },
+      applicationProcessing: {
+        status: status || undefined,
+      },
+      isRenewal: requestType ? requestType === 'Renewal' : undefined,
+      permitType: permitType || undefined,
+      AND: [
+        {
+          OR: [
+            { firstName: { contains: firstSearch, mode: 'insensitive' } },
+            { middleName: { contains: middleSearch, mode: 'insensitive' } },
+            { lastName: { contains: lastSearch, mode: 'insensitive' } },
+          ],
+        },
+      ],
+    },
+  });
+
+  // Get applications with filter, sorting, pagination
+  const applications = await prisma.application.findMany({
+    skip: offset,
+    take: limit,
+    orderBy: orderBy,
+    where: {
+      applicant: {
+        id: userIDSearch,
+      },
+      applicationProcessing: {
+        status: status || undefined,
+      },
+      isRenewal: requestType ? requestType === 'Renewal' : undefined,
+      permitType: permitType || undefined,
+      AND: [
+        {
+          OR: [
+            { firstName: { contains: firstSearch, mode: 'insensitive' } },
+            { middleName: { contains: middleSearch, mode: 'insensitive' } },
+            { lastName: { contains: lastSearch, mode: 'insensitive' } },
+          ],
+        },
+      ],
+    },
+    select: {
+      firstName: true,
+      lastName: true,
+      id: true,
+      createdAt: true,
+      permitType: true,
+      isRenewal: true,
+      applicationProcessing: {
+        select: {
+          status: true,
```

Comment:
Is there a reason we're explicitly selecting only these columns? I don't think we should constrain which fields are being returned. Graphql already handles selecting only the data that we need in the query.

Reply:
Good point. we had to specifically select the applicationProcessing field but I realized we can use `include` instead so I switched to that